evadb/executor/create_function_executor.py

```diff
@@ -245,253 +282,280 @@ def handle_ultralytics_function(self):
 
     def handle_forecasting_function(self):
         """Handle forecasting functions"""
-        os.environ["CUDA_VISIBLE_DEVICES"] = ""
-        aggregated_batch_list = []
-        child = self.children[0]
-        for batch in child.exec():
-            aggregated_batch_list.append(batch)
-        aggregated_batch = Batch.concat(aggregated_batch_list, copy=False)
-        aggregated_batch.drop_column_alias()
-
-        arg_map = {arg.key: arg.value for arg in self.node.metadata}
-        if not self.node.impl_path:
-            impl_path = Path(f"{self.function_dir}/forecast.py").absolute().as_posix()
-        else:
-            impl_path = self.node.impl_path.absolute().as_posix()
-        library = "statsforecast"
-        supported_libraries = ["statsforecast", "neuralforecast"]
-
-        if "horizon" not in arg_map.keys():
-            raise ValueError(
-                "Horizon must be provided while creating function of type FORECASTING"
-            )
-        try:
-            horizon = int(arg_map["horizon"])
-        except Exception as e:
-            err_msg = f"{str(e)}. HORIZON must be integral."
-            logger.error(err_msg)
-            raise FunctionIODefinitionError(err_msg)
+        cuda_devices_here = "0"
+        if "CUDA_VISIBLE_DEVICES" in os.environ:
+            cuda_devices_here = os.environ["CUDA_VISIBLE_DEVICES"].split(",")[0]
+
+        with set_env(CUDA_VISIBLE_DEVICES=cuda_devices_here):
+            aggregated_batch_list = []
```

Comment:
If I understand correctly, only the neueralforcast fit needs to change the `CUDA_VISISBLE_DEVICES`.  Can we move the `set_env` only around the neuralforcast's `model.fit`? Thanks!